assets/app/views/project.html

```diff
@@ -11,38 +11,45 @@
+                   service, these are likely new so incoming builds are likely, so lets show these now.
+               -->
+              <div ng-repeat="(deploymentConfigId, deploymentConfig) in serviceDeploymentConfigs[serviceId]" ng-if="!serviceDeploymentsByDeploymentConfig[serviceId][deploymentConfigId] && deploymentConfigs[deploymentConfigId]">
+                <!-- As long as pods created from the deployments created from this deployment config will be routed
+                     to by this service, then we will show things related to triggers that are about to create stuff, like builds.
+                -->                
+                <triggers ng-init="triggers = deploymentConfigs[deploymentConfigId].triggers"></triggers>                
+              </div>
+              <!-- Now go through all deployments for this service grouped by deployment config (or lack thereof) -->
+              <div ng-repeat="(deploymentConfigId, deployments) in serviceDeploymentsByDeploymentConfig[serviceId]">
                 <!-- deploymentConfig could be null when we have any deployments that were not generated from a deploymentConfig -->
-                <div ng-if="deploymentConfig">
-                  <div ng-repeat="trigger in deploymentConfig.triggers">
-                    <div ng-repeat="build in trigger.builds" style="border-bottom: 1px solid #ddd; padding-bottom: 10px; margin-top: 10px;" ng-if="(build.status != 'Complete' && build.status != 'Failed') || (build.metadata.creationTimestamp | ageLessThan : 5 : 'minutes')" class="animate-repeat">
-                      <span ng-switch="build.status" class="hide-ng-leave">
-                        <span ng-switch-when="Complete" class="fa fa-check text-success" aria-hidden="true" style="margin-right: 5px;"></span>
-                        <span ng-switch-when="Failed" class="fa fa-times text-danger" aria-hidden="true" style="margin-right: 5px;"></span>
-                        <span ng-switch-default class="fa fa-refresh fa-spin" aria-hidden="true" style="margin-right: 5px;"></span>
-                      </span>
-                      <span>A build of {{build.metadata.labels.buildconfig}} is {{build.status | lowercase}}.</span>
-                      <span ng-if="build.status != 'Complete' && build.status != 'Failed' && trigger.imageChangeParams.automatic">  A new deployment will be created automatically once the build completes.</span>
-                    </div>
-                  </div>
+                <!-- As long as pods created from the deployments created from this deployment config will be routed
+                     to by this service, then we will show things related to triggers that are about to create stuff, like builds.
+                -->
+                <div ng-if="deploymentConfigs[deploymentConfigId] && serviceDeploymentConfigs[serviceId][deploymentConfigId]">
```

Comment:
can you remove the `deploymentConfigs[deploymentConfigId]` check and just do `ng-if="deploymentConfigId && serviceDeploymentConfigs[serviceId][deploymentConfigId]"`